R/get_awards.R

```diff
@@ -0,0 +1,101 @@
+#' Get NSF Arctic/Polar program award information
+#'
+#' Uses the [NSF API](https://www.research.gov/common/webapi/awardapisearch-v1.htm)
+#' to get all records pertaining to the Arctic or Polar programs.
+#'
+#' @param from_date (character) Optional. Returns all
+#' records with start date after specified date.
+#' Format = "mm/dd/yyyy"
+#' @param query (character) Optional. By default, the function
+#' searches for all awards with either "polar" or "arctic" in
+#' the fundProgramName. Additional queries can be specified
+#' as defined in the NSF API.
+#' Use '&' to join multiple queries (i.e., "keyword=water&agency=NASA")
+#' @param print_fields (character) Optional. By default, the
+#' following fields will be returned: id, date,
+#' startDate, expDate, fundProgramName, poName,
+#' title, awardee, piFirstName, piLastName, piPhone, piEmail
+#'
+#' @import XML
+#' @import stringr
+#' @import RCurl
+#'
+#' @export
+#'
+#' @author Irene Steves
+#'
+#' @examples
+#' \dontrun{
+#' all_awards <- get_awards()
+#' new_awards <- get_awards(from_date = "01/01/2017")
+#' }
+
+get_awards <- function(from_date = NULL,
+                       query = NULL,
+                       print_fields = NULL) {
+
+    # package checks
+    if (!requireNamespace("XML")) {
+        stop(call. = FALSE,
+             "The XML package is required for this function. Please install it and try again.")
+    }
+    if (!requireNamespace("stringr")) {
+        stop(call. = FALSE,
+             "The stringr package is required for this function. Please install it and try again.")
+    }
+    if (!requireNamespace("RCurl")) {
+        stop(call. = FALSE,
+             "The RCurl package is required for this function. Please install it and try again.")
+    }
+
+    # basic argument checks
+    stopifnot(is.character(from_date) | is.null(from_date))
+    stopifnot(is.character(query) | is.null(query))
+    stopifnot(is.character(print_fields) | is.null(print_fields))
+
+    base_url <- "https://api.nsf.gov/services/v1/awards.xml?fundProgramName=ARCTIC|fundProgramName=POLAR"
+    if(!is.null(query)) {
+        query <- paste0("&", query)
+    }
+
+    if(is.null(print_fields)) {
```

Comment:
What if you made these fields always be returned and changed `print_fields` to something like `additional_fields` where the user could specify additional fields to return?  That way if they wanted to customize the fields returned they wouldn't need to type all of these in.

Reply:
interesting idea...

Reply:
After thinking about it, I won't change for now, but might in the future